backend/app/core/review_helpers.py

```
@@ -0,0 +1,90 @@
+    if len(reviews) <= sample_size:
+        return reviews
+
+    # rating 별 그룹핑
+    by_rating: dict[int, list[dict]] = {}
+    for r in reviews:
+        key = int(r.get("metadata", {}).get("rating", r.get("rating", 0)))
+        by_rating.setdefault(key, []).append(r)
+
+    sampled: list[dict] = []
+    total = len(reviews)
+    for rating, group in by_rating.items():
```

Ruff (0.15.12)

[warning] Loop control variable `rating` not used within loop body

Rename unused `rating` to `_rating`

(B007)